File: src/WebUI/src/app/modules/pokemon/services/pokemon.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PaginatedList } from '../models/paginated-list';
import { Pokemon } from '../models/pokemon.model';
import { Specie } from '../models/specie.model';

@Injectable()
export class PokemonService {
  private readonly baseUrl = 'http://localhost:5268';

  constructor(private http: HttpClient) {}

  public async getPokemon(): Promise<Observable<any>> {
    // this.store
    //   .collection('pokemons')
    //   .valueChanges()
    //   .subscribe((result) => {
    //     // console.log(result);
    //     return result;
    //   });
    return this.http.get(`${this.baseUrl}/pokemon`);
  }

  public async getPokemonWithPagination(
    page: number,
    pageSize: number
  ): Promise<Observable<PaginatedList<Pokemon>>> {
    return this.http.get<PaginatedList<Pokemon>>(
      `${this.baseUrl}/pokemonWithPagination?pageNumber=${page}&pageSize=${pageSize}`
    );
  }

  public async getPokemonByExternalId(
    externalId: number
  ): Promise<Observable<Pokemon>> {
    return this.http.get<Pokemon>(
      `${this.baseUrl}/pokemonByExternalId?externalId=${externalId}`
    );
  }

  public async getPokemonSpecieByExternalId(
    externalId: number
  ): Promise<Observable<Specie>> {
    return this.http.get<any>(
      `${this.baseUrl}/pokemonDescriptionByExternalId?externalId=${externalId}`
    );
  }
}

File: src/App/Services/PokemonService.cs
using App.Helpers;
using App.Models;

namespace App.Services
{
    public interface IPokemonService
    {
        Task<GenericResponse<Pokemon>> GetPokemon(int pageNumber, int pageSize);
    }

    public class PokemonService : IPokemonService
    {
        public PokemonService()
        {

        }

        public async Task<GenericResponse<Pokemon>> GetPokemon(int pageNumber, int pageSize)
        {
            //var response = await HttpHelper.GetAsync<GenericResponse<Pokemon>>($"http://10.0.2.2:5268/pokemonWithPagination?pageNumber={pageNumber}&pageSize={pageSize}");
            var response = await HttpHelper.GetAsync<GenericResponse<Pokemon>>($"http://localhost:5268/pokemonWithPagination?pageNumber={pageNumber}&pageSize={pageSize}");
            return response;
        }
    }
}
